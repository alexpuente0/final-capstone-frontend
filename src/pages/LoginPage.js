/* eslint-disable react/jsx-props-no-spreading */
import { useEffect } from 'react';
import { useForm } from 'react-hook-form';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate, useLocation } from 'react-router-dom';
import { login } from '../redux/auth/auth';
import './login.css';

function LoginPage() {
  const dispatch = useDispatch();
  const { register, handleSubmit } = useForm();

  const authUser = useSelector((state) => state.auth.user);
  const authError = useSelector((state) => state.auth.error);

  const navigate = useNavigate();
  const location = useLocation();

  useEffect(() => {
    // redirect to from if already logged in
    if (authUser) {
      const { from } = location.state || { from: { pathname: '/' } };
      navigate(from.pathname);
    }
  }, [authUser, location.state, navigate]);

  function onSubmit({ email, password }) {
    return dispatch(login(email, password));
  }

  return (
    <div className="loginpage addreservpage">
      <h1>Login</h1>
      <div className="centered">
        <form className="myform" onSubmit={handleSubmit(onSubmit)}>
          <div className="form-group">
            <input className="formitem" type="text" {...register('email')} id="email" name="email" placeholder="Email" required minLength="2" maxLength="30" />
          </div>
          <div className="form-group">
            <input className="formitem" type="password" {...register('password')} id="password" name="password" placeholder="Password" required />
          </div>
          <div className="button-wrapper">
            &nbsp;
            <button className="sbutton" type="submit">Submit</button>
            &nbsp;
          </div>
          {authError && <div className="alert alert-danger mt-3 mb-0">{authError.message}</div>}
        </form>
      </div>
    </div>
  );
}

export default LoginPage;
