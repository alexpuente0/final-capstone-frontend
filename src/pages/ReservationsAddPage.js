/* eslint-disable react/jsx-props-no-spreading */
import { useEffect, useState } from 'react';
import { useForm } from 'react-hook-form';
import { useDispatch, shallowEqual, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getItems } from '../redux/item/itemReducer';
import { addReservation } from '../redux/reserv/reserv';
import history from '../helpers/history';
import './addReserv.css';

function ReservationsAddPage() {
  const dispatch = useDispatch();
  const { id } = useParams();

  useEffect(() => {
    dispatch(getItems());
  }, [dispatch]);

  const [redirect, setRedirect] = useState(false);
  useEffect(() => {
    // redirect to reservations after successful submit
    if (redirect) {
      history.navigate('/reservations');
    }
  }, [redirect]);

  const cars = useSelector(
    (state) => state.items.items.filter((item) => (!id || item.id.toString() === id)), shallowEqual,
  ) || [];

  const { register, handleSubmit } = useForm();

  function onSubmit({ item, city, date }) {
    setRedirect(true);
    return dispatch(addReservation(item, city, date));
  }

  function MySelect() {
    return (
      <select className="formitem" id="item" {...register('item')} name="item" placeholder="Car" required>
        {cars.map((car) => (
          <option key={car.name} value={car.name}>{car.name}</option>
        ))}
      </select>
    );
  }

  return (
    <div className="addreservpage">
      <h1>Add Reservation</h1>
      <div className="centered">
        <form className="myform2" onSubmit={handleSubmit(onSubmit)}>
          <div className="form-group">
            {MySelect()}
          </div>
          <div className="form-group">
            <input className="formitem" type="text" {...register('city')} id="city" name="city" placeholder="City" required minLength="2" maxLength="30" />
          </div>
          <div className="form-group">
            <input className="formitem" type="date" {...register('date')} id="date" name="date" placeholder="Date" required />
          </div>
          <div className="button-wrapper">
            &nbsp;
            <button className="sbutton" type="submit">Submit</button>
            &nbsp;
          </div>
        </form>
      </div>
    </div>
  );
}

export default ReservationsAddPage;
