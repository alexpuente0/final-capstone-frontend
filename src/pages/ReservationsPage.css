.reservcontainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0 10px 10px 10px;
  min-height: 100vh;
  width: 100vw;
  background-size: cover;
}

.reservpage {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.reservtable {
  margin: 10px;
  padding: 0;
}

.listitem {
  padding: 5px;
  list-style: none;
  display: flex;
  justify-content: space-between;
}

.listitem:nth-child(even) {
  background-color: white;
}

.listitem:nth-child(odd) {
  background-color: #5f942b7c;
}

.listitem:nth-child(1) {
  border-radius: 5px;
  border: 0;
  background-color: #07ae81;
  font-weight: bold;
}

.listitem:last-child {
  border-radius: 5px;
  border: 0;
}

.tablewrapper {
  border: 1px solid black;
  border-radius: 5px;
  margin: 10px;
  width: 90%;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.spancentered {
  text-align: center;
  width: 30%;
}

.fixedwidth {
  width: 30%;
}

.smallfont {
  font-size: 0.875rem;
}

@media only screen and (min-width: 800px) {
  .tablewrapper {
    width: 70%;
  }
}
