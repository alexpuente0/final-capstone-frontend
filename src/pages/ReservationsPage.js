import { useEffect } from 'react';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import { getReserv } from '../redux/reserv/reserv';
import './ReservationsPage.css';

function ReservationsPage() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getReserv());
  }, [dispatch]);

  const reservations = useSelector((state) => state.reservations.reservations, shallowEqual) || [];
  const myreserv = reservations.map((reservation) => {
    const d = new Date(reservation.date);
    const reserv = { ...reservation, date: d.toLocaleDateString() };
    return reserv;
  });

  return (
    <>
      <h1>Reservations Page</h1>
      <div className="tablewrapper">
        <ul className="reservtable">
          <li className="listitem">
            <span className="fixedwidth">Car</span>
            <span className="spancentered">City</span>
            <span className="spancentered">Date</span>
          </li>
          {myreserv.map((reservation) => (
            <li key={reservation.id} className="listitem">
              <span className="fixedwidth">{reservation.carname}</span>
              <span className="spancentered">{reservation.city}</span>
              <span className="spancentered smallfont">{reservation.date}</span>
            </li>
          ))}
        </ul>
      </div>
    </>
  );
}

export default ReservationsPage;
