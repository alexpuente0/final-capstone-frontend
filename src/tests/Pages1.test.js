import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import LoginPage from '../pages/LoginPage';
import ReservationsPage from '../pages/ReservationsPage';
import ReservationsAddPage from '../pages/ReservationsAddPage';
import payload from './__mocks__/reserv_mock';

test('LoginPage renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <LoginPage />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('ReservationsAddPage renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <ReservationsAddPage />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('ReservationsPage renders correctly for mock data', () => {
  store.dispatch({ type: 'GET_RESERV', payload });
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <ReservationsPage />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
