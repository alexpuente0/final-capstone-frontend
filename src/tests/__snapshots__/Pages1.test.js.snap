// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders correctly 1`] = `
<div
  className="loginpage"
>
  <h1>
    Login
  </h1>
  <div
    className="centered"
  >
    <form
      className="myform"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <input
          className="formitem"
          id="email"
          maxLength="30"
          minLength="2"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="formitem"
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="button-wrapper"
      >
         
        <button
          className="sbutton"
          type="submit"
        >
          Submit
        </button>
         
      </div>
    </form>
  </div>
</div>
`;

exports[`ReservationsAddPage renders correctly 1`] = `
<div
  className="addreservpage"
>
  <h1>
    Add Reservation
  </h1>
  <div
    className="centered"
  >
    <form
      className="myform2"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <select
          className="formitem"
          id="item"
          name="item"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Car"
          required={true}
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="formitem"
          id="city"
          maxLength="30"
          minLength="2"
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="City"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="formitem"
          id="date"
          name="date"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Date"
          required={true}
          type="date"
        />
      </div>
      <div
        className="button-wrapper"
      >
         
        <button
          className="sbutton"
          type="submit"
        >
          Submit
        </button>
         
      </div>
    </form>
  </div>
</div>
`;

exports[`ReservationsPage renders correctly for mock data 1`] = `
Array [
  <h1>
    Reservations Page
  </h1>,
  <div
    className="tablewrapper"
  >
    <ul
      className="reservtable"
    >
      <li
        className="listitem"
      >
        <span
          className="fixedwidth"
        >
          Car
        </span>
        <span
          className="spancentered"
        >
          City
        </span>
        <span
          className="spancentered"
        >
          Date
        </span>
      </li>
      <li
        className="listitem"
      >
        <span
          className="fixedwidth"
        >
          Genesis GV60
        </span>
        <span
          className="spancentered"
        >
          Mexico DF
        </span>
        <span
          className="spancentered smallfont"
        >
          20/9/2022
        </span>
      </li>
      <li
        className="listitem"
      >
        <span
          className="fixedwidth"
        >
          Kia EV6
        </span>
        <span
          className="spancentered"
        >
          Buenos Aires
        </span>
        <span
          className="spancentered smallfont"
        >
          1/6/2022
        </span>
      </li>
      <li
        className="listitem"
      >
        <span
          className="fixedwidth"
        >
          BMW i4
        </span>
        <span
          className="spancentered"
        >
          San Luis
        </span>
        <span
          className="spancentered smallfont"
        >
          11/10/2022
        </span>
      </li>
    </ul>
  </div>,
]
`;
